### PR TITLE
kasli: add SED lanes count option to HW description JSON file

### PR DESCRIPTION
**Pull request by airwoodix**
# ARTIQ Pull Request

## Description of Changes

This patch adds an entry to the JSON hardware description file for Kasli to configure the number of FIFOs in the SED.

Moderately increasing the number of lanes (from 8 to 16 lanes) seems to be unproblematic and greatly helps mitigating sequence errors. The JSON schema sets the limit to 32, since this is the maximum I got to build using Vivado 2020.1.

An extension of this patch may be to expose more gateware configuration options in the JSON file, but this is probably very error prone.

## Type of Changes

|   | Type |
| ------------- | ------------- |
|  | :bug: Bug fix  |
| ✓  | :sparkles: New feature |
|   | :hammer: Refactoring  |
|   | :scroll: Docs |

### All Pull Requests

- [x] Use correct spelling and grammar.
- [x] Update [RELEASE_NOTES.rst](../RELEASE_NOTES.rst) if there are noteworthy changes, especially if there are changes to existing APIs.
- [x] Check the copyright situation of your changes and sign off your patches (`git commit --signoff`, see [copyright](../CONTRIBUTING.rst#copyright-and-sign-off)).

### Code Changes

- [x] Run `flake8` to check code style (follow PEP-8 style). `flake8` has issues with parsing Migen/gateware code, ignore as necessary.
- [x] Test your changes or have someone test them. Mention what was tested and how.

### Git Logistics

- [x] Split your contribution into logically separate changes (`git rebase --interactive`). Merge/squash/fixup commits that just fix or amend previous commits. Remove unintended changes & cleanup. See [tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
- [x] Write short & meaningful commit messages. Review each commit for messages (`git show`). Format:
  ```
  topic: description. < 50 characters total.
  
  Longer description. < 70 characters per line
  ```

### Licensing

See [copyright & licensing for more info](https://github.com/m-labs/artiq/blob/master/CONTRIBUTING.rst#copyright-and-sign-off).
ARTIQ files that do not contain a license header are copyrighted by M-Labs Limited and are licensed under LGPLv3+.

### CODE REVIEW

**Comment by han94ros** [comment]
I am currently having problems with sequence errors on the urukul and zotino SPI channels, however increasing the number of SED lanes to 32 has not solved the issue. I have also tried setting the enable_spread to false. Do you know which other changes I need to make to the ARTIQ gateware.

File: artiq/coredevice/coredevice_generic.schema.json [hunk 1]
```diff
@@ -64,6 +64,13 @@
             "type": "boolean",
             "default": false
         },
+	"sed_lanes": {
+	    "type": "number",
+	    "minimum": 1,
+	    "maximum": 32,
+	    "default": 8,
+	    "description": "Number of FIFOs in the SED, must be a power of 2"
+	},
```
Comment (sbourdeauducq) [inline artiq/coredevice/coredevice_generic.schema.json:73]: Indentation looks wrong.

Comment (airwoodix) [reply artiq/coredevice/coredevice_generic.schema.json]: Indeed, sorry. Wrong editor configuration... Should be fixed in 1b719a1

**Comment by sbourdeauducq** [comment]
What about DRTIO devices?

**Comment by airwoodix** [comment]
Good point, thanks. Does fcc7ab8 fix this appropriately? I don't have hardware to test the satellite part.

**Comment by airwoodix** [comment]
@sbourdeauducq is this a patch you'd be willing to merge?

**Pull request merged**